### ISSUE TITLE
Automatically create cpanfile somehow (with version numbers)

### ISSUE THREAD

**Issue by akarelas** [perl-carton/carton#176]
``` bash
npm install <plugin> --save-devs
```

This places the plugin in `packages.json`, together with the version number.

I think it's better than having to type `cpanfile` character by character.

**Comment by systocrat** [comment]
Why is this considered a silly request? Almost every major language has a package manager with a feature like this. Python has pip freeze, Node has the npm snippet you mentioned, even Ruby has an equivalent with bundle.

I'm not as familiar with Perl as I should be and I'm not trying to patronize- I'm legitimately wondering why this shouldn't be considered as a feature.

**Comment by miyagawa** [comment]
Ruby's bundler doesn't have this, you always update Gemfile first, then running `bundle` will create the lock file, like Carton/Carmel works.

**Comment by systocrat** [comment]
Ah, well- I'm not as familiar with Ruby as I should be either then :)
Either way, I'm not sure I'm seeing why this shouldn't be implemented here-
Unless cpan or cpanm should be the software handling it?

**Comment by miyagawa** [comment]
I do not consider this a silly request - though because Module::CPANfile exists, you can easily write a command that adds the requested module to the `cpanfile`, save and then run `carton install`.

**Comment by melo** [comment]
Just another data point.

What @miyagawa suggests is what we did internally at work. For background, we use git + git submodules, and each git repo has a separate cpanfile. We wrote a util with two subcommands:

* add: searches the closest cpanfile  and adds the module to it, it defaults to the latest version found on metacpan;
* merge: scans the entire project for cpanfiles, and merges them into a single one at the root of the project.

It currently assumes Carton (it has switches to run carton install automatically), but we are also experimenting with Carmel.

This works very well for us, project side has around 200 direct deps, and Carton installs 450 dists.
